File: MaximumConsecutiveOnes/maxConsecutiveOnes.js
class Solution {
    findMaxConsecutiveOnes(nums) {
        let cnt = 0;
        let maxi = 0;

        for (let i = 0; i < nums.length; i++) {
            if (nums[i] == 1) {
                cnt++;
                maxi = Math.max(maxi, cnt);

            } else {
                cnt = 0;
            }
        }
        return maxi;
    }
}

const nums = [1, 1, 0, 1, 1, 1];
const sol = new Solution();

const ans = sol.findMaxConsecutiveOnes(nums);
console.log("The maximum consecutive 1's are " + ans);

File: MaximumConsecutiveOnes/MaxConsecutiveOnes.cs
using System;

class Program
{
    public static void Main(string[] args) {
        int[] nums = {1, 1, 1, 1, 1}; 
        int result = FindMaxConsecutiveOnes(nums);
        Console.WriteLine("Maximum consecutive ones: " + result);
    }
    public static int FindMaxConsecutiveOnes(int[] nums) {
        int maxCount = 0;
        int currentCount = 0;

        foreach (int num in nums) {
            if (num == 1) {
                currentCount++;
                maxCount = Math.Max(maxCount, currentCount);
            } else {
                currentCount = 0;
            }
        }

        return maxCount;
    }
}
